pyabc/sampler/redis_eps/sampler.py

```diff
@@ -270,37 +319,19 @@ def clear_generation_t(self, t: int) -> None:
         """
         ana_id = self.analysis_id
         # delete keys from pipeline
-        pipeline = self.redis.pipeline()
-        pipeline.delete(idfy(SSA, ana_id, t))
-        pipeline.delete(idfy(N_EVAL, ana_id, t))
-        pipeline.delete(idfy(N_ACC, ana_id, t))
-        pipeline.delete(idfy(N_REQ, ana_id, t))
-        pipeline.delete(idfy(N_FAIL, ana_id, t))
-        pipeline.delete(idfy(ALL_ACCEPTED, ana_id, t))
-        pipeline.delete(idfy(N_WORKER, ana_id, t))
-        pipeline.delete(idfy(BATCH_SIZE, ana_id, t))
-        pipeline.delete(idfy(QUEUE, ana_id, t))
-        pipeline.execute()
-
-    def create_sample(self, id_results: List[Tuple], n: int) -> Sample:
-        """Create a single sample result.
-        Order the results by starting point to avoid a bias towards
-        short-running simulations (dynamic scheduling).
-        """
-        # sort
-        id_results.sort(key=lambda x: x[0])
-        # cut off
-        id_results = id_results[:n]
-
-        # extract simulations
-        results = [res[1] for res in id_results]
-
-        # create 1 to-be-returned sample from results
-        sample = self._create_empty_sample()
-        for j in range(n):
-            sample += results[j]
-
-        return sample
+        (self.redis.pipeline()
+         .delete(idfy(SSA, ana_id, t))
+         .delete(idfy(N_EVAL, ana_id, t))
+         .delete(idfy(N_ACC, ana_id, t))
+         .delete(idfy(N_REQ, ana_id, t))
+         .delete(idfy(N_FAIL, ana_id, t))
+         .delete(idfy(ALL_ACCEPTED, ana_id, t))
+         .delete(idfy(N_WORKER, ana_id, t))
+         .delete(idfy(BATCH_SIZE, ana_id, t))
+         .delete(idfy(IS_LOOK_AHEAD, ana_id, t))
+         .delete(idfy(MODE, ana_id, t))
+         .delete(idfy(QUEUE, ana_id, t))
+         .execute())
```

Comment:
How come the only thing happening in the sample_until_n_accepted function is adding and deleting to the pipeline?

Reply:
Nevermind, i just missed the button to open the rest ^^

Reply:
But still, all those pipeline.set and pipeline.delete parts used to be in a separate function, didnt they?

Reply:
They are actually. GitHub just displays the difference very weirdly. They should be in start_generation_t and clear_generation_t functions if I remember correctly.